components/service/contile/src/main/java/mozilla/components/service/contile/ContileTopSitesProvider.kt

```diff
@@ -4,32 +4,70 @@
 
 package mozilla.components.service.contile
 
+import android.content.Context
+import android.util.AtomicFile
+import androidx.annotation.VisibleForTesting
 import mozilla.components.concept.fetch.Client
 import mozilla.components.concept.fetch.Request
 import mozilla.components.concept.fetch.isSuccess
 import mozilla.components.feature.top.sites.TopSite
 import mozilla.components.feature.top.sites.TopSitesProvider
 import mozilla.components.support.base.log.logger.Logger
 import mozilla.components.support.ktx.android.org.json.asSequence
+import mozilla.components.support.ktx.util.readAndDeserialize
+import mozilla.components.support.ktx.util.writeString
 import org.json.JSONException
 import org.json.JSONObject
+import java.io.File
 import java.io.IOException
+import java.util.Date
 
 internal const val CONTILE_ENDPOINT_URL = "https://contile.services.mozilla.com/v1/tiles"
+internal const val CACHE_FILE_NAME = "mozilla_components_service_contile.json"
+internal const val MINUTE_IN_MS = 60 * 1000
 
 /**
  * Provide access to the Contile services API.
  *
+ * @property context A reference to the application context.
  * @property client [Client] used for interacting with the Contile HTTP API.
+ * @property endPointURL The url of the endpoint to fetch from. Defaults to [CONTILE_ENDPOINT_URL].
+ * @property maxCacheAgeInMinutes Maximum time (in minutes) the cache should remain valid
+ * before a refresh is attempted. Defaults to -1, meaning no cache is being used by default.
  */
 class ContileTopSitesProvider(
-    private val client: Client
+    private val context: Context,
+    private val client: Client,
+    private val endPointURL: String = CONTILE_ENDPOINT_URL,
+    private val maxCacheAgeInMinutes: Long = -1
 ) : TopSitesProvider {
 
     private val logger = Logger("ContileTopSitesProvider")
 
+    private val diskCacheLock = Any()
```

Comment:
for my edification, is this dummy variable used to enforce synchronous caching? Is it necessary because of our caching implementation?